mlrun/datastore/azure_blob.py

```diff
@@ -64,75 +65,94 @@ def get_storage_options(self):
             sas_token=self._get_secret_or_env("AZURE_STORAGE_SAS_TOKEN"),
         )
 
+    def _convert_key_to_remote_path(self, key):
+        key = key.strip("/")
+        path = Path(self.endpoint, key).as_posix()
+        return path
+
     def upload(self, key, src_path):
         if self.bsc:
             # Need to strip leading / from key
-            blob_client = self.bsc.get_blob_client(
+            with self.bsc.get_blob_client(
                 container=self.endpoint, blob=key[1:]
-            )
-            with open(src_path, "rb") as data:
-                blob_client.upload_blob(data, overwrite=True)
+            ) as blob_client:
+                with open(src_path, "rb") as data:
+                    blob_client.upload_blob(data, overwrite=True)
         else:
-            remote_path = f"{self.endpoint}{key}"
-            self._filesystem.put_file(src_path, remote_path)
+            remote_path = self._convert_key_to_remote_path(key)
+            self._filesystem.put_file(src_path, remote_path, overwrite=True)
 
     def get(self, key, size=None, offset=0):
         if self.bsc:
-            blob_client = self.bsc.get_blob_client(
+            with self.bsc.get_blob_client(
                 container=self.endpoint, blob=key[1:]
-            )
-            size = size if size else None
-            return blob_client.download_blob(offset, size).readall()
+            ) as blob_client:
+                size = size if size else None
+                blob = blob_client.download_blob(offset, size).readall()
+                return blob
         else:
-            blob = self._filesystem.cat_file(key, start=offset)
+            remote_path = self._convert_key_to_remote_path(key)
+            blob = self._filesystem.cat_file(remote_path, start=offset)
```

Comment:
Continuing our discussion from previous PR - what if you pass `end = offset + size` to `cat_file` if size is not None? Will adlfs code handle it gracefully if `end` is past EOF? 
I think if that works, it's better to implement that (otherwise `size` is ignored in this code). I suggest something like this:

```suggestion
            end = offset + size if size else None
            blob = self._filesystem.cat_file(remote_path, start=offset, end=end)
```

Reply:
@theSaarco adlfs's behavior will mirror what would occur if using AZURE_STORAGE_CONNECTION_STRING to authenticate, and the user passed a "size" parameter that exceeded the size of the blob that was being retrieved.